javalib/src/main/scala/java/io/Reader.scala

```diff
@@ -16,13 +16,15 @@ import java.nio.CharBuffer
 
 import scala.annotation.tailrec
 
-abstract class Reader private[this] (_lock: Option[Object])
-    extends Readable with Closeable {
-
-  protected val lock = _lock.getOrElse(this)
-
-  protected def this(lock: Object) = this(Some(lock))
-  protected def this() = this(None)
+abstract class Reader() extends Readable with Closeable {
+  protected var lock: Object = this
+
+  protected def this(lock: Object) = {
```

Comment:
Have you considered making this the primary ctor? (Assuming `this(this)` is allowed).

Having a `var` instead of a `val` in the entire class' scope seems a rather big price to pay for an extra NPE check.

Reply:
Unfortunately, `this(this)` is not allowed. You can only use `this` *after* the primary constructor call.

Reply:
> Having a `var` instead of a `val` in the entire class' scope seems a rather big price to pay for an extra NPE check.

It's actually supposed to be a `var`. This is a non-final `protected` field in the Java API:
https://docs.oracle.com/javase/8/docs/api/java/io/Reader.html#lock